[0.6 documentation] Fix Page Developer Tools: Bundled Program

zhenyan-zhang-meta opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #10194

https://github.com/pytorch/executorch/issues/10193

- Update the hyperlink of “Generate Sample ExecuTorch program”, as this part has been heavily revamped in v0.6:
  - Update to https://pytorch.org/executorch/0.6/getting-started.html#exporting
- Fixed the emit stage sample code:
  - Use `register_buffer` as warning suggests.
  - Update `to_edge` to be `to_edge_transform_and_lower`, as updated in https://pytorch.org/executorch/0.6/getting-started.html.
- Remove function blocks for `get_program_data`, `load_bundled_input` and `load_bundled_output`, as there are no other similar usage in the doc.
  - Changed to links of real-world examples.
- Remove hard-coded cpp code examples.
  - For APIs, provide links to real-world examples.
  - For runtime example, change to a ready-to-use snippet for people to quickly try out.

Differential Revision: [D73027728](https://our.internmc.facebook.com/intern/diff/D73027728/)

Commit:
[0.6 documentation] Fix Page Developer Tools: Bundled Program

https://github.com/pytorch/executorch/issues/10193

- Update the hyperlink of “Generate Sample ExecuTorch program”, as this part has been heavily revamped in v0.6:
  - Update to https://pytorch.org/executorch/0.6/getting-started.html#exporting
- Fixed the emit stage sample code:
  - Use `register_buffer` as warning suggests.
  - Update `to_edge` to be `to_edge_transform_and_lower`, as updated in https://pytorch.org/executorch/0.6/getting-started.html.
- Remove function blocks for `get_program_data`, `load_bundled_input` and `load_bundled_output`, as there are no other similar usage in the doc.
  - Changed to links of real-world examples.
- Remove hard-coded cpp code examples.
  - For APIs, provide links to real-world examples.
  - For runtime example, change to a ready-to-use snippet for people to quickly try out.

Differential Revision: [D73027728](https://our.internmc.facebook.com/intern/diff/D73027728/)

[ghstack-poisoned]

Gasoonjia commented on docs/source/bundled-io.md:
```diff
@@ -105,8 +105,8 @@ class SampleModel(torch.nn.Module):
         super().__init__()
-        self.a: torch.Tensor = 3 * torch.ones(2, 2, dtype=torch.int32)
-        self.b: torch.Tensor = 2 * torch.ones(2, 2, dtype=torch.int32)
+        self.register_buffer('a', 3 * torch.ones(2, 2, dtype=torch.int32))
```
y use register_buffer to replace torch.tensor?

zhenyan-zhang-meta replied:
To solve warnings like `UserWarning: Node a target a a of  does not reference an nn.Module, nn.Parameter, or buffer, which is what 'get_attr' Nodes typically target`

Gasoonjia commented on docs/source/bundled-io.md:
````diff
@@ -199,103 +198,29 @@ This stage mainly focuses on executing the model with the bundled inputs and and
+```bash
+cd executorch
+   ./examples/devtools/build_example_runner.sh
+   ./cmake-out/examples/devtools/example_runner --bundled_program_path {your-bpte-file}.bpte --output_verification
````
```suggestion
   ./cmake-out/examples/devtools/example_runner --bundled_program_path {your-bpte-file} --output_verification
```
i think only bracelet here should be fine

Gasoonjia commented on docs/source/bundled-io.md:
````diff
@@ -199,103 +198,29 @@ This stage mainly focuses on executing the model with the bundled inputs and and
+   ./cmake-out/examples/devtools/example_runner --bundled_program_path {your-bpte-file}.bpte --output_verification
 ```
 
+It is expected to see no input.
````
What's that mean?

zhenyan-zhang-meta replied:
It is expected to see no input from running the above mentioned snippet.

Let me update the wording.

Gasoonjia replied:
hmmm i‘m still confused here: what do you mean by "no input to see"? What's the expected or hypnosis input user want to see?

zhenyan-zhang-meta replied:
Sheesh I mean no output 😅

Gasoonjia reviewed (commented):
Thanks for updating! Overall looks good and left subtle comments

Commit:
Update on "[0.6 documentation] Fix Page Developer Tools: Bundled Program"


https://github.com/pytorch/executorch/issues/10193

- Update the hyperlink of “Generate Sample ExecuTorch program”, as this part has been heavily revamped in v0.6:
  - Update to https://pytorch.org/executorch/0.6/getting-started.html#exporting
- Fixed the emit stage sample code:
  - Use `register_buffer` as warning suggests.
  - Update `to_edge` to be `to_edge_transform_and_lower`, as updated in https://pytorch.org/executorch/0.6/getting-started.html.
- Remove function blocks for `get_program_data`, `load_bundled_input` and `load_bundled_output`, as there are no other similar usage in the doc.
  - Changed to links of real-world examples.
- Remove hard-coded cpp code examples.
  - For APIs, provide links to real-world examples.
  - For runtime example, change to a ready-to-use snippet for people to quickly try out.

Differential Revision: [D73027728](https://our.internmc.facebook.com/intern/diff/D73027728/)

[ghstack-poisoned]

zhenyan-zhang-meta commented:
@Gasoonjia Thanks for reviewing! Updated as suggested.

Commit:
Update on "[0.6 documentation] Fix Page Developer Tools: Bundled Program"


https://github.com/pytorch/executorch/issues/10193

- Update the hyperlink of “Generate Sample ExecuTorch program”, as this part has been heavily revamped in v0.6:
  - Update to https://pytorch.org/executorch/0.6/getting-started.html#exporting
- Fixed the emit stage sample code:
  - Use `register_buffer` as warning suggests.
  - Update `to_edge` to be `to_edge_transform_and_lower`, as updated in https://pytorch.org/executorch/0.6/getting-started.html.
- Remove function blocks for `get_program_data`, `load_bundled_input` and `load_bundled_output`, as there are no other similar usage in the doc.
  - Changed to links of real-world examples.
- Remove hard-coded cpp code examples.
  - For APIs, provide links to real-world examples.
  - For runtime example, change to a ready-to-use snippet for people to quickly try out.

Differential Revision: [D73027728](https://our.internmc.facebook.com/intern/diff/D73027728/)

[ghstack-poisoned]

Gasoonjia reviewed (approved):
LGTM! Thanks!
Please cherry pick this PR.

zhenyan-zhang-meta commented:
@pytorchbot cherry-pick --onto release/0.6 -c doc

zhenyan-zhang-meta commented:
@pytorchbot cherry-pick --onto release/0.6 -c docs
